cpp/src/parquet/encoding.cc

```diff
@@ -3177,8 +3194,7 @@ void DeltaByteArrayEncoder<DType>::Put(const T* src, int num_values) {
                                           ::arrow::stl::allocator<int32_t>(pool_));
   std::string_view last_value_view = last_value_;
 
-  int i = 0;
-  while (i < num_values) {
+  for (int i = 0; i < num_values; i++) {
```

Comment:
nit: what about make the current function a specialization for ByteArrayType and throw an unimplemented exception in the generic one? This may make the line 3199 less confusing.

Reply:
Added generic function that throws and changed the comments a little bit.